server/handler/simulate.go

```diff
@@ -0,0 +1,182 @@
+// Copyright 2018 Palantir Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package handler
+
+import (
+	"context"
+	"net/http"
+	"strings"
+
+	"github.com/palantir/go-baseapp/baseapp"
+	"github.com/palantir/go-githubapp/githubapp"
+	"github.com/palantir/policy-bot/policy"
+	"github.com/palantir/policy-bot/policy/common"
+	"github.com/palantir/policy-bot/policy/simulated"
+	"github.com/palantir/policy-bot/pull"
+	"github.com/palantir/policy-bot/server/apierror"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog"
+)
+
+// Simulate provides a baseline for handlers to perform simulated pull request evaluations and
+// either return the result or display it in the ui.
+type Simulate struct {
+	Base
+}
+
+// SimulationResponse is the response returned from Simulate, this is a trimmed down version of common.Result with json
+// tags. This struct and the newSimulationResponse constructor can be extended to include extra content from common.Result.
+type SimulationResponse struct {
+	Name              string `json:"name"`
+	Description       string `json:"description:"`
+	StatusDescription string `json:"status_description"`
+	Status            string `json:"status"`
+	Error             string `json:"error"`
+}
+
+func (h *Simulate) ServeHTTP(w http.ResponseWriter, r *http.Request) error {
+	ctx := r.Context()
+	logger := *zerolog.Ctx(ctx)
+
+	token := getToken(r)
+	if token == "" {
+		return apierror.WriteAPIError(w, http.StatusUnauthorized, "missing token")
+	}
+
+	client, err := h.NewTokenClient(token)
+	if err != nil {
+		logger.Error().Msg("failed to create token client")
+		return apierror.WriteAPIError(w, http.StatusUnauthorized, "token invalid")
+	}
+
+	owner, repo, number, ok := parsePullParams(r)
+	if !ok {
+		logger.Error().Msg("failed to parse pull request parameters from request")
+		return apierror.WriteAPIError(w, http.StatusBadRequest, "failed to parse pull request parameters from request")
+	}
+
+	pr, _, err := client.PullRequests.Get(ctx, owner, repo, number)
+	if err != nil {
+		if isNotFound(err) {
+			logger.Error().Err(err).Msg("could not find pull request")
+			return apierror.WriteAPIError(w, http.StatusNotFound, "failed to find pull request")
+		}
+
+		return errors.Wrap(err, "failed to get pr")
```

Comment:
nit:
```suggestion
		return errors.Wrap(err, "failed to get pull request")
```
for consistency with the other messages